### PR TITLE
Replace componentWillReceiveProps with componentDidUpdate to fix warnings on react 16.8

### LINKED ISSUE #195: in the latest version of react we get the following warning due to the use of deprecated APIs. Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.

### PR DESCRIPTION
**Pull request by jeancx**
closes #195
```
Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.
    
    * Move data fetching code or side effects to componentDidUpdate.
    * If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
    * Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.
    
    Please update the following components: AutosizeInput, TagsInput
```
More info:
https://pt-br.reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -420,7 +420,7 @@ class TagsInput extends React.Component {
     })
   }
 
-  componentWillReceiveProps (nextProps) {
+  componentDidUpdate (nextProps) {
```
Comment (jeancx) [inline src/index.js]:
```suggestion
  componentDidUpdate (prevProps) {
```

File: src/index.js [hunk 2]
```diff
@@ -430,9 +430,9 @@ class TagsInput extends React.Component {
       return
     }
 
-    this.setState({
-      tag: this.inputValue(nextProps)
-    })
+    if (this.inputValue(nextProps) !== this.inputValue(this.props)) {
+      this.setState({ tag: this.inputValue(nextProps) })
```
Comment (jeancx) [inline src/index.js]:
```suggestion
      this.setState({ tag: this.inputValue(this.props) })
```

File: src/index.js [hunk 3]
```diff
@@ -430,9 +430,9 @@ class TagsInput extends React.Component {
       return
     }
 
-    this.setState({
-      tag: this.inputValue(nextProps)
-    })
+    if (this.inputValue(nextProps) !== this.inputValue(this.props)) {
```
Comment (jeancx) [inline src/index.js]:
```suggestion
    if (this.inputValue(prevProps) !== this.inputValue(this.props)) {
```

**Comment by lourenci** [comment]
Hi, is there any plan to merge this?

**Comment by Zmeu213** [comment]
@lourenci looks like this repo is abandoned 
@olahol are you alive?

**Comment by lorenzopolidori** [comment]
@jeancx shall we for this and publish independently?

**Comment by abhishekbha007** [comment]
Can we merge this?

**Pull request closed without merging**